python/cuml/ensemble/randomforestregressor.pyx

```diff
@@ -0,0 +1,620 @@
+                   Ratio of number of features (columns) to consider
+                   per node split.
+                   If int then max_features/n_features.
+                   If float then max_features is a fraction.
+                   If 'auto' then max_features=n_features which is 1.0.
+                   If 'sqrt' then max_features=1/sqrt(n_features).
+                   If 'log2' then max_features=log2(n_features)/n_features.
+                   If None, then max_features=n_features which is 1.0.
+    n_bins :  int (default = 8)
+              Number of bins used by the split algorithm.
+    min_rows_per_node : int or float (default = 2)
+                        The minimum number of samples (rows) needed
+                        to split a node.
+                        If int then number of sample rows
+                        If float the min_rows_per_sample*n_rows
+    accuracy_metric : string (default = 'mse')
+                      Decides the metric used to evaluate the performance
+                      of the model.
+                      'median'
+    """
+
+    variables = ['n_estimators', 'max_depth', 'handle',
+                 'max_features', 'n_bins',
+                 'split_algo', 'split_criterion', 'min_rows_per_node',
+                 'bootstrap', 'bootstrap_features',
+                 'verbose', 'rows_sample',
+                 'max_leaves', 'quantile_per_tree',
+                 'accuracy_metric']
+
+    def __init__(self, n_estimators=10, max_depth=-1, handle=None,
+                 max_features=None, n_bins=8,
+                 split_algo=1, split_criterion=2,
+                 bootstrap=True, bootstrap_features=False,
+                 verbose=False, min_rows_per_node=2,
+                 rows_sample=1.0, max_leaves=-1,
+                 accuracy_metric='mse', min_samples_leaf=None,
+                 min_weight_fraction_leaf=None, n_jobs=None,
+                 max_leaf_nodes=None, min_impurity_decrease=None,
+                 min_impurity_split=None, oob_score=None,
+                 random_state=None, warm_start=None, class_weight=None,
+                 quantile_per_tree=False, criterion=None):
+
+        sklearn_params = {"criterion": criterion,
+                          "min_samples_leaf": min_samples_leaf,
+                          "min_weight_fraction_leaf": min_weight_fraction_leaf,
+                          "max_leaf_nodes": max_leaf_nodes,
+                          "min_impurity_decrease": min_impurity_decrease,
+                          "min_impurity_split": min_impurity_split,
+                          "oob_score": oob_score, "n_jobs": n_jobs,
+                          "random_state": random_state,
+                          "warm_start": warm_start,
+                          "class_weight": class_weight}
+
+        for key, vals in sklearn_params.items():
+            if vals is not None:
+                raise TypeError(" The sklearn variable ", key,
```

Comment:
```suggestion
                raise TypeError(" The Scikit-learn variable ", key,
```

Reply:
Also it would be good to extend this type of warning/error to all our classes, could you open an issue?

Reply:
sure!